Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSonicWallFirewall.yaml

```diff
@@ -0,0 +1,432 @@
+  Version: '0.1'
+  LastUpdated: October 16, 2023
+Product:
+  Name: SonicWall
+Normalization:
+  Schema: NetworkSession
+  Version: '0.2.2'
+References:
+  - Title: ASIM Network Session Schema
+    Link: https://aka.ms/ASimNetworkSessionDoc
+  - Title: ASIM
+    Link: https://aka.ms/AboutASIM
+Description: |
+  This ASIM parser supports normalizing SonicWall SonicOS ArcSight-formatted Syslog data ingested by Microsoft Sentinel into CommonSecurityLogs to the ASIM Network Session normalized schema.
+ParserName: ASimNetworkSessionSonicWallFirewall
+EquivalentBuiltInParser: _ASim_NetworkSession_SonicWallFirewall
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let Actions=datatable(fw_action:string,DvcAction:string)
+  [ "reset client","Reset Source"
+  , "reset server","Reset Destination"
+  , "reset both", "Reset" 
+  , "allow","Allow"
+  , "\"forward\"","Allow"
+  , "\"mgmt\"","Management"
+  , "\"NA\"","NA"
+  , "deny","Deny"
+  , "\"drop\"", "Drop"
+  , "drop ICMP", "Drop ICMP"];
+  let Parser=(disabled:bool=false){
+  CommonSecurityLog
+  | where not(disabled)
+  | where DeviceVendor == "SonicWall"
+  | where ( isnotempty(SourceIP) or isnotempty(DestinationIP) )
+  | invoke _ASIM_ResolveDstFQDN("DestinationHostname")
+  | invoke _ASIM_ResolveSrcFQDN("SourceHostname")
+  | invoke _ASIM_ResolveDvcFQDN("")
+  | parse-kv AdditionalExtensions as (['gcat']:string, ['snpt']:string, ['dnpt']:string, ['susr']:string,['appName']:string, ['appcat']:string, ['appid']:string, ['sid']:string, ['catid']:string, ['ipscat']:string, ['ipspri']:string, ['spycat']:string, ['spypri']:string, ['fw_action']:string, ['dpi']:string, ['bid']:string, ['af_action']:string, ['af_polid']:string, ['af_policy']:string, ['af_type']:string, ['af_service']:string, ['af_object']:string, ['contentObject']:string, ['fileid']:string, ['uuid']:string) with (pair_delimiter=";", kv_delimiter="=")
+  | extend RequestURL_ = extract(@"(?:[.*;]+?)(?P<url>[a-zA-Z0-9_*.,}{&%$~:;\-=\/?[:space:]]+)(?:;fw_action)", 1, AdditionalExtensions)
+  | extend RequestURL_ = iif(RequestURL_ startswith "snpt" or RequestURL_ startswith "dnpt" or RequestURL_ startswith "appid" or RequestURL_ startswith "appName", extract(@"(?:\d;|.{1}\w.{1};)(?P<url>[a-zA-Z0-9_*.,}{&%$~:;\-=\/?[:space:]]+)", 1, RequestURL_), RequestURL_)
+  | extend RequestURL_ = iif(RequestURL_ matches regex @"^(.{2,6}=.{1,6})", extract(@"(?:\d;|.{1}\w.{1};)(?P<url>[a-zA-Z0-9_*.,}{&%$~:;\-=\/?[:space:]]+)", 1, RequestURL_), iif(RequestURL_ matches regex @"^\w=\d$", "", RequestURL_))
+  | lookup Actions on fw_action
+  // Sets the mandatory EventResult based on the DvcAction.
+  | extend EventResult = case(DvcAction == "Allow", "Success",
+                            DvcAction == "Management", "NA",
+                            DvcAction == "NA", "NA",
+                            "Failure"
+                            )
+  | extend LogMsgSeverity = case(LogSeverity == 10, "Emergency (0)",

```

Comment:
not in network session schema. do we need this? You are already mapping EventSeverity

Reply:
The event severities used by Sentinel differ in values and have a different ordering. My thought process was to include the severity based on the friendly string we use in SonicOS. Is that OK, or do I need to remove it?

Reply:
I see now that I can use AdditionalFields to include additional details. I'm making the necessary changes to remove anything not in schema.

Reply:
sounds good